src/Nethermind/Nethermind.Runner/Ethereum/Api/ApiBuilder.cs

```diff
@@ -31,31 +32,29 @@ public ApiBuilder(IConfigProvider configProvider, ILogManager logManager)
         _configProvider = configProvider ?? throw new ArgumentNullException(nameof(configProvider));
         _initConfig = configProvider.GetConfig<IInitConfig>();
         _jsonSerializer = new EthereumJsonSerializer();
+        ChainSpec = LoadChainSpec(_jsonSerializer);
     }
 
     public INethermindApi Create(params IConsensusPlugin[] consensusPlugins) =>
         Create((IEnumerable<IConsensusPlugin>)consensusPlugins);
 
     public INethermindApi Create(IEnumerable<IConsensusPlugin> consensusPlugins)
     {
-        ChainSpec chainSpec = LoadChainSpec(_jsonSerializer);
         bool wasCreated = Interlocked.CompareExchange(ref _apiCreated, 1, 0) == 1;
         if (wasCreated)
         {
             throw new NotSupportedException("Creation of multiple APIs not supported.");
         }
 
-        string engine = chainSpec.SealEngineType;
-        IConsensusPlugin? enginePlugin = consensusPlugins.FirstOrDefault(p => p.SealEngineType == engine);
-
+        IConsensusPlugin? enginePlugin = consensusPlugins.FirstOrDefault();
```

Comment:
no need to discriminate by engine type?

Reply:
Still not sure, but each plugin seems too be exclusive to each other. They do exactly the same check, just check with chainSpec.sealEngineType.

Reply:
I'd say, do a check tto make sure only one IConsensusPlugin is on.